src/Core/Resolvers/Factories/QueryManagerFactory.cs

```diff
@@ -0,0 +1,109 @@
+        {
+            _queryBuilders = new Dictionary<DatabaseType, IQueryBuilder>();
+            _queryExecutors = new Dictionary<DatabaseType, IQueryExecutor>();
+            _dbExceptionsParsers = new Dictionary<DatabaseType, DbExceptionParser>();
+            foreach (DataSource dataSource in runtimeConfigProvider.GetConfig().ListAllDataSources())
+            {
+                IQueryBuilder? queryBuilder = null;
+                IQueryExecutor? queryExecutor = null;
+                DbExceptionParser? exceptionParser = null;
+
+                if (_queryBuilders.ContainsKey(dataSource.DatabaseType))
+                {
+                    // we have already created the builder, parser and executor for this database type.No need to create again.
+                    continue;
+                }
+
+                switch (dataSource.DatabaseType)
+                {
+                    case DatabaseType.CosmosDB_NoSQL:
+                        break;
+                    case DatabaseType.MSSQL:
+                        queryBuilder = new MsSqlQueryBuilder();
+                        exceptionParser = new MsSqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new MsSqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    case DatabaseType.MySQL:
+                        queryBuilder = new MySqlQueryBuilder();
+                        exceptionParser = new MySqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new MySqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    case DatabaseType.PostgreSQL:
+                        queryBuilder = new PostgresQueryBuilder();
+                        exceptionParser = new PostgreSqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new PostgreSqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    default:
+                        throw new NotSupportedException(dataSource.DatabaseTypeNotSupportedMessage);
+                }
+
+                _queryBuilders.TryAdd(dataSource.DatabaseType, queryBuilder!);
+                _queryExecutors.TryAdd(dataSource.DatabaseType, queryExecutor!);
+                _dbExceptionsParsers.TryAdd(dataSource.DatabaseType, exceptionParser!);
+            }
+        }
+
+        /// <inheritdoc />
+        public IQueryBuilder GetQueryBuilder(DatabaseType databaseType)
+        {
+            if (!_queryBuilders.TryGetValue(databaseType, out IQueryBuilder? queryBuilder))
+            {
+                throw new DataApiBuilderException($"{nameof(DatabaseType)}:{databaseType} could not be found within the config", HttpStatusCode.BadRequest, DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
```

Comment:
split line here and other similar places.

Reply:
fixed.